src/content/blog/2022-08-05-new-config-system-part-2.md

```diff
@@ -0,0 +1,512 @@
+---
+layout: post
+title: "ESLint's new config system, Part 2: Introduction to flat config"
+teaser: "ESLint's new config system, nicknamed flat config, is designed to be both familiar and much simpler than the original config system."
+tags:
+  - Configuration
+  - Command Line
+authors:
+  - nzakas
+categories:
+  - API Changes
+---
+
+In [my previous post](https://eslint.org/blog/2022/08/new-config-system-part-1/), I talked about how the eslintrc config system had grown to be more complex than necessary through a series of small, incremental changes. The flat config system, on the other hand, was designed from the start to be simpler in a number of ways. We took all of the learnings from the previous six years of ESLint development to come up with a holistic approach to configuration that took the best of eslintrc and combined it with the way other JavaScript-related tools handled configuration. The result is something that hopefully feels familiar to existing ESLint users and is far more powerful than what was possible before.
+
+**Docs:** Read more about flat config system in the [official documentation](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new).
+
+## The goals of flat config
+
+To set the stage for the changes in flat config, we had several goals:
+
+1. **Logical defaults** - the way people write JavaScript has changed a lot in the past nine years, and we wanted the new config system to reflect our current reality rather than the one we lived in when ESLint was first released.
+1. **One way to define configs** - we didn't want folks to have multiple ways to do the same thing any longer. There should be one way to define configs for any given project.
+1. **Rules configs should remain unchanged** - we felt like the way rules were configured already worked fine, so to make it easier to transition to flat config, we didn't want to make any changes to rule configs. The same `rules` key can be used the same way in flat config.
+1. **Use native loading for everything** - one of our biggest regrets about eslintrc was recreating the Node.js `require` resolution in a custom way. This was a significant source of complexity and, in hindsight, unnecessary. Going forward, we wanted to leveral the loading capabilities of the JavaScript runtime directly.
+1. **Better organized top-level keys** - the number of keys at the top-level of eslintrc had grown drammatically since ESLint was released. We need to look at which keys were necessary and how they related to one another.
```

Comment:
```suggestion
1. **Better organized top-level keys** - the number of keys at the top-level of eslintrc had grown dramatically since ESLint was released. We need to look at which keys were necessary and how they related to one another.
```